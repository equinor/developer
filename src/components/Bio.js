import React, { Component } from "react";
import kebabCase from "lodash/kebabCase";
import styled from "styled-components";
import Image from "gatsby-image";

const Wrapper = styled.div`
  display: flex;
  margin-bottom: 0;
  flex-direction: row;
  @media (max-width: 576px) {
    flex-direction: column;
    margin-bottom: 20px;
  }
`;

const AuthorContent = styled.div`
  display: flex;
  margin-right: 10px;
  flex-direction: column;
`;

const Avatar = styled.img`
  margin-bottom: 0;
  margin-right: 10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
`;

class Authors extends Component {
  render() {
    const { authors } = this.props;

    const list = authors && authors.map((author, index) => {
      return <Bio key={`${author}-${index}`} author={author} />;
    });

    return <Wrapper>{list}</Wrapper>;
  }
}

export { Authors };

class Bio extends Component {
  render() {
    const { author } = this.props;

    const { name, image } = author;

    return (
      <Wrapper>
        <Avatar
          src={image}
          alt={name}
          imgStyle={{
            borderRadius: `50%`,
          }}
        />
        <AuthorContent>
          <div className="bio-name">{`${name}`}</div>
          {false && (
            <div className="bio-email">
              <a href={`mailto:${email}`}>{email}</a>
            </div>
          )}
          {false && <div className="bio-bio">{bio}</div>}
        </AuthorContent>
      </Wrapper>
    );
  }
}

export default Bio;
