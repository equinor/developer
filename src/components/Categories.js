import React from "react";
import { Link } from "gatsby";
import styled from "styled-components";
import { Col, Row } from "react-styled-flexboxgrid";
import { getIcon } from "../util/iconUtil";

const Categories = props => {
  const { nodes } = props;

  const CategoryTitle = styled.div`
    color: #333333;
    font-family: Equinor;
    font-size: 30px;
    letter-spacing: -0.08px;
    line-height: 36px;
    margin-top: 5px;
    margin-bottom: 15px;
  `;

  const iconStyle = {
    margin: 0,
    width: "auto",
    height: "auto",
  };

  const categoryComponents = nodes.map(({ node }) => {
    const {
      frontmatter: { title, tags },
      fields: { slug, collection },
    } = node;
    const Icon = getIcon(tags[0]);
    return (
      <Col
        md={4}
        sm={6}
        xs={12}
        key={"category-" + node.fields.slug}
        style={{ marginTop: 50 }}
      >
        <Icon style={iconStyle} />
        <CategoryTitle>
          <Link to={`/${collection}${slug}`}>{title}</Link>
        </CategoryTitle>
        <div>{node.excerpt}</div>
      </Col>
    );
  });

  return <Row>{categoryComponents}</Row>;
};

export { Categories };
