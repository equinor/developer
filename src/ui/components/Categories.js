import React from "react";
import { graphql, Link } from "gatsby";
import styled from "styled-components";
import ApiIcon from "../../assets/icons/Api.svg";
import OpenSourceIcon from "../../assets/icons/OpenSource.svg";
import TechIcon from "../../assets/icons/Tech.svg";
import SecurityIcon from "../../assets/icons/Security.svg";
import DesignIcon from "../../assets/icons/Design.svg";
import { Col, Row } from "react-styled-flexboxgrid";

const Categories = props => {
  const { categories } = props;

  const CategoryTitle = styled.div`
    color: #333333;
    font-family: Equinor;
    font-size: 30px;
    letter-spacing: -0.08px;
    line-height: 36px;
    margin-top: 5px;
    margin-bottom: 15px;
  `;

  const iconStyle = {
    margin: 0,
    width: "auto",
    height: "auto",
  };

  const categoryComponents = categories.map((category, index) => {
    const Icon = getIcon(category.slug);

    return (
      <Col
        md={4}
        sm={6}
        xs={12}
        key={"category-" + category.slug}
        style={{ marginTop: 50 }}
      >
        <Icon style={iconStyle} />
        <CategoryTitle>
          <Link to={"/docs/" + category.slug}>{category.title}</Link>
        </CategoryTitle>
        <div>{category.description}</div>
      </Col>
    );
  });

  return <Row>{categoryComponents}</Row>;
};

export { Categories };

function getIcon(slug) {
  switch (slug) {
    case "/api/":
      return ApiIcon;
    case "/open-source/":
      return OpenSourceIcon;
    case "/tech/":
      return TechIcon;
    case "/security/":
      return SecurityIcon;
    case "/design/":
      return DesignIcon;
    default:
      throw `icon type ${slug} is not supported. `;
  }
}
