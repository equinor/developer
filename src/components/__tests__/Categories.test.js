import React from "react";
import renderer from "react-test-renderer";
import { Categories } from "../Categories";
import { mockAuthors } from "./Bio.test";

describe("Categories", () => {
  const node = {
    frontmatter: { title: "Api", tags: ["Api"] },
    fields: { authors: mockAuthors },
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Categories nodes={[{ node }]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
