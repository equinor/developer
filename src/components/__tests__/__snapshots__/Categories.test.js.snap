// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  margin-bottom: 0;
  margin-right: 10px;
  width: 40px;
  height: 40px;
  border-radius: 100%;
}

@media (max-width:576px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-bottom: 20px;
  }
}

<div
  className="c0"
>
  <div
    className="c0"
  >
    <img
      alt="Ola Nordmann"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Ola Nordmann
      </div>
    </div>
  </div>
  <div
    className="c0"
  >
    <img
      alt="Kari Nordmann"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Kari Nordmann
      </div>
    </div>
  </div>
  <div
    className="c0"
  >
    <img
      alt="Erik Eriksen"
      className="c1"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAaQAAAGkCAIAAADxLsZiAAAF4klEQVR4nOzXAY3jMBBA0dtTiJRWyEQhE1qhYAjLoeraav57BGakxl/TbYzxD+Dp/q9eAGAGsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxK2aZOu8542i7ftx2v1Cp/kq/sKc746lx2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkDCNm3SfrymzZrjOu/VK1D0vKc0h8sOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5I+BljrN4B4M+57IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IGGbNuk672mzptmP1+oVaPGO3uayAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDEsQOSBA7IEHsgASxAxLEDkgQOyBB7IAEsQMSxA5IEDsgQeyABLEDErbVC3y367xXr/Bh+/FavcInPe8H4m3zYvewV+QhscrzntIc/sYCCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJP2OM1TsA/DmXHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkCB2QILYAQliBySIHZAgdkCC2AEJYgckiB2QIHZAgtgBCWIHJIgdkPAbAAD//9JGJL1D6GKKAAAAAElFTkSuQmCC"
    />
    <div
      className="c2"
    >
      <div
        className="bio-name"
      >
        Erik Eriksen
      </div>
    </div>
  </div>
</div>
`;

exports[`Categories renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: -0.5rem;
  margin-left: -0.5rem;
}

.c1 {
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding-right: 0.5rem;
  padding-left: 0.5rem;
}

.c2 {
  color: #333333;
  font-family: Equinor;
  font-size: 30px;
  -webkit-letter-spacing: -0.08px;
  -moz-letter-spacing: -0.08px;
  -ms-letter-spacing: -0.08px;
  letter-spacing: -0.08px;
  line-height: 36px;
  margin-top: 5px;
  margin-bottom: 15px;
}

@media only screen and (min-width:0em) {
  .c1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    max-width: 100%;
    display: block;
  }
}

@media only screen and (min-width:48em) {
  .c1 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    max-width: 50%;
    display: block;
  }
}

@media only screen and (min-width:64em) {
  .c1 {
    -webkit-flex-basis: 33.333333333333336%;
    -ms-flex-preferred-size: 33.333333333333336%;
    flex-basis: 33.333333333333336%;
    max-width: 33.333333333333336%;
    display: block;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <test-file-stub
      style={
        Object {
          "height": "auto",
          "margin": 0,
          "width": "auto",
        }
      }
    />
    <div
      className="c2"
    >
      <a
        href="/undefinedundefined"
      >
        Api
      </a>
    </div>
    <div />
  </div>
</div>
`;
